app/src/main/java/me/capcom/smsgateway/ui/settings/MessagesSettingsFragment.kt

```
@@ -45,6 +46,25 @@ class MessagesSettingsFragment : BasePreferenceFragment() {
+
+            "messages.work_hours_start", "messages.work_hours_end" -> {
+                val pref = preference as EditTextPreference
+                val currentText = pref.text ?: "09:00"
+                val parts = currentText.split(":")
+                val hour = parts.getOrNull(0)?.toIntOrNull()?.coerceIn(0, 23) ?: 9
+                val minute = parts.getOrNull(1)?.toIntOrNull()?.coerceIn(0, 59) ?: 0
+
+                TimePickerDialog(
+                    requireContext(),
+                    { _, h, m ->
+                        pref.text = String.format("%02d:%02d", h, m)
```

detekt (1.23.8)

[warning] String.format("%02d:%02d", h, m) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)